### PR TITLE
feat: always request revealed addresses for joining communities

### PR DESCRIPTION
**Pull request by MishkaRogachev**
Waits https://github.com/status-im/status-go/pull/3569

For https://github.com/status-im/status-desktop/pull/10967
Issue: https://github.com/status-im/status-desktop/issues/10853

Important changes:
- [x] Always request revealed addresses for joining communities

### CODE REVIEW

**Comment by MishkaRogachev** [comment]
waits for https://github.com/status-im/status-go/pull/3569

**Comment by MishkaRogachev** [comment]
Moving it from draft for reviews, but it requires cleanup and rebasing after https://github.com/status-im/status-go/pull/3569 get merged

**Comment by MishkaRogachev** [comment]
Upd: draft until next week

**Comment by 0x-r4bbit** [comment]
> Moving it from draft for reviews, but it requires cleanup and rebasing after https://github.com/status-im/status-go/pull/3569 get merged

@MishkaRogachev actually, this PR does not need to be based on top of admin permissions. I think in fact you'd be better off just rebasing it on top of latest `develop`.

**Comment by MishkaRogachev** [comment]
> > Moving it from draft for reviews, but it requires cleanup and rebasing after #3569 get merged
> 
> @MishkaRogachev actually, this PR does not need to be based on top of admin permissions. I think in fact you'd be better off just rebasing it on top of latest `develop`.

I've created a new PR for develop: https://github.com/status-im/status-go/pull/3609

File: account/accounts.go [hunk 1]
```diff
@@ -61,6 +61,11 @@ type RecoverParams struct {
 	Message   string `json:"message"`
 	Signature string `json:"signature"`
 }
+type AccountInterface interface {
+	GetVerifiedWalletAccount(db *accounts.Database, address, password string) (*SelectedExtKey, error)
+	Sign(rpcParams SignParams, verifiedAccount *SelectedExtKey) (result types.HexBytes, err error)
+	Recover(rpcParams RecoverParams) (addr types.Address, err error)
+}
```
Comment (Samyoul) [inline account/accounts.go:68]: Nice use of interface. Would you be able to add the go docs for this?

```suggestion

// AccountInterface represents something and is used for things
type AccountInterface interface {
	GetVerifiedWalletAccount(db *accounts.Database, address, password string) (*SelectedExtKey, error)
	Sign(rpcParams SignParams, verifiedAccount *SelectedExtKey) (result types.HexBytes, err error)
	Recover(rpcParams RecoverParams) (addr types.Address, err error)
}
```

**Comment by Samyoul** [comment]
@MishkaRogachev thank you for this work. I'm in the middle of reviewing this, but one thing that stands out to me is this PR is named `always request revealed addresses for joining communities`, but it seems to be a major refactor of the community permissions system.

Does this PR need to be rebased against another branch?

File: account/accounts.go (hunk 1 above)
Comment (MishkaRogachev) [reply account/accounts.go:68]: Sure! Btw this PR on hold until next week

**Comment by MishkaRogachev** [comment]
@Samyoul  @0x-r4bbit tests and mock move to this PR: https://github.com/status-im/status-go/pull/3609/. You can review that instead :)

**Comment by Samyoul** [comment]
@MishkaRogachev does this PR need to be closed now?

**Comment by MishkaRogachev** [comment]
> @MishkaRogachev does this PR need to be closed now?

Yes, i plan to close it after https://github.com/status-im/status-go/pull/3637 get merged

**Pull request closed without merging**